### PR TITLE
[BugFix] Support concrete scalar types as return column types of polymorphic TVFs

### PR DESCRIPTION
**Pull request by agubichev**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

IF the TVF is declared as returning both polymorphic column types and concrete types, the concrete types right now are dropped when resolving the type.
For example:

FUNC(ANY_ARRAY) -> <ANY_ELEMENT, BIGINT> is the template,

and the invocation is 
FUNC(ARRAY_VARCHAR).

During resolution, we correctly resolve ANY_ELEMENT as VARCHAR, but we also want to retain BIGINT as one of the return columns.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PolymorphicFunctionAnalyzer.java|1|1|100.00%|[]|

File: fe/fe-core/src/test/java/com/starrocks/catalog/FunctionSetTest.java [hunk 1]
```diff
@@ -242,4 +243,29 @@ public void testPolymorphicFunction() {
...
+        // First two columns of the result are polymorphic (derived from argument type), but the last argument is of a
+        // "concrete" type BIGINT, which is retained in the resolved function.
+        TableFunction polymorphicTVF =
+                new TableFunction(new FunctionName("three_column_tvf"), Lists.newArrayList("a", "b", "c"),
+                        Lists.newArrayList(Type.ANY_ARRAY),
+                        Lists.newArrayList(Type.ANY_ELEMENT, Type.ANY_ELEMENT, Type.BIGINT));
+
+        functionSet.addBuiltin(polymorphicTVF);
+
+
```
Comment (zuyu) [inline fe/fe-core/src/test/java/com/starrocks/catalog/FunctionSetTest.java:258]: Remove this empty line.

Does two column tvf also work?

Comment (agubichev) [reply fe/fe-core/src/test/java/com/starrocks/catalog/FunctionSetTest.java]: added a test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PolymorphicFunctionAnalyzer.java|1|1|100.00%|[]|

**Review by imay — DISMISSED** [review]

**Comment by imay** [comment]
@agubichev you need to sign off by 'git commit -s'

**Review by kangkaisen — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by imay — APPROVED** [review]

**Comment by imay** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by imay** [comment]
@mergify backport branch-3.0

**Comment by imay** [comment]
@agubichev This patch has a conflict in branch-2.5. If you want it in 2.5, would you mind cherry-picking it to branch 2.5 manually?

**Pull request merged**